---
title: "Analysis of the effect of TorsinA RNAi on proteotoxicity expressing C. elegans"
description: "Analysis of the CEL-Seq2 results received by the Technion"
author: "Amir Levine"
date: "2018-10-29"
output: md_document
---

**2019-05-20 Stopped before Gene tables.**

# Pre-code

## Set parameters
```{r parameters, results='hide'}
today = Sys.Date()
results_dir = "../results/"
data_dir = "../data/"
```

## Packages
Download and install. Only once. Packages defined in the next cell.
```{r import-packages, results='hide'}
install.packages(packages) # Install packages
```

```{r load-packages}
packages <- c(
  "BiocManager","tidyverse",
  "VennDiagram","viridis", # Venn Diagram
  "gplots","RColorBrewer", # Heatmap plot
  "biomaRt","org.Ce.eg.db", # Gene Annotations
  "fgsea", # GSEA
  "Cairo", # Export PDF/SVG
  "ggpubr" # Publication-quality plots arrangement
)

lapply(packages, require, character.only = TRUE) # Load packages
```


# Data Cleaning

## Import data

Import the gene expression data sent by the Technion. Specifiy how to treat NA values. Specify the columns' data-types, since the default loads all as characters. 
```{r import-data}
AM <- readxl::read_excel(
  path = "raw/2018-10-25-Results_files/Final_results_AM140_treated_vs_untreated_.xlsx", 
  col_names = TRUE, 
  col_types = c(rep("text",4),rep("numeric",5),rep("text",4)), 
  na = 'NA')
CL <- readxl::read_excel(
  path = "raw/2018-10-25-Results_files/Final_results_CL2006_treated_vs_untreated_.xlsx",
  col_names = TRUE, 
  col_types = c(rep("text",4),rep("numeric",5),rep("text",4)), 
  na = 'NA')
N2 <- readxl::read_excel(
  path = "raw/2018-10-25-Results_files/Final_results_N2_treated_vs_untreated.xlsx",
  col_names = TRUE, 
  col_types = c(rep("text",4),rep("numeric",5),rep("text",4)), 
  na = 'NA')
```

## Merge strains (all protein coding genes)
Merge the three strain specific dataframes to one gene expression (ge) dataframe. Remove unnecessary columns for further analysis. Focus on protein coding genes, regardless of whether they underwent a statistical text. Sort the dataframe by fold-change.
```{r process-data}
gepc_read <- 
  bind_rows(list(N2 = N2,AM140 = AM, CL2006 = CL), .id = 'strain') %>%
  dplyr::filter(`flag Treated vs untreated` != "All_Zero") %>% 
  # All positive reads
  dplyr::filter(`gene biotype` == "protein_coding") %>% 
  # Ignore mapping to non-protein coding genes
  dplyr::rename(Gene_Name = `gene name`,
                Gene_ID = `Gene ID`,
                log2FC = `log2FoldChange Treated vs untreated`,
                pvalue = `pvalue Treated vs untreated`,
                FDR_p = `padj Treated vs untreated`,
                flag = `flag Treated vs untreated`) %>%
  dplyr::select(strain,
                Gene_ID,
                Gene_Name,
                baseMean,
                log2FC,
                pvalue,
                FDR_p,
                flag) %>%
  arrange(log2FC) %>%
  group_by(strain)
```

## Filter tested genes only

Focus on genes that underwent the multiple comparisons test (*Tested*). Sort the dataframe by the adjusted p-value.
```{r process-data}
gepc_tested <- gepc_read %>%
  dplyr::filter(flag == "Tested") %>% # High counts only.
  arrange(FDR_p)
```


## Save clean data
Save cleaned dataframes and remove individual strains data-frames (save checkpoint).
```{r save-data}
save(AM,CL,N2,gepc_read,gepc_tested,
     file = "data/import-clean-ge.Rdata")
rm(list = c("AM","CL","N2"))
```

## Generate Gene Lists (Up-/Down-regulated)
Used in Venn diagram and Heatmap

### Load Dataframes (**checkpoint**).
```{r load-data}
# load("data/import-clean-ge.Rdata")
```

Obtain gene lists of upregulated genes (*p*-value < 0.1), for each strain. Create a gene-name vector to generate a heatmap and save the data-frame.
```{r get-upregulated}
AM <- gepc_tested %>% 
  filter(strain == "AM140",
         FDR_p < 0.1,
         log2FC > 0) %>%
  pull(`Gene ID`)

CL <- gepc_tested %>% 
  filter(strain == "CL2006",
         FDR_p < 0.1,
         log2FC > 0) %>%
  pull(`Gene ID`)

N2 <- gepc_tested %>% 
  filter(strain == "N2",
         FDR_p < 0.1,
         log2FC > 0) %>%
  pull(`Gene ID`)

all.up <- unique(c(AM,CL,N2))

save(AM,CL,N2,all.up,file = "data/up.Rdata")
```

Obtain gene lists of downregulated genes (*p*-value < 0.1), for each strain.
Combine all genes for heatmap.
```{r get-downregulated}
AM <- gepc_tested %>% 
  filter(strain == "AM140",
         FDR_p < 0.1,
         log2FC < 0) %>%
  pull(`Gene ID`)

CL <- gepc_tested %>% 
  filter(strain == "CL2006",
         FDR_p < 0.1,
         log2FC < 0) %>%
  pull(`Gene ID`)

N2 <- gepc_tested %>% 
  filter(strain == "N2",
         FDR_p < 0.1,
         log2FC < 0) %>%
  pull(`Gene ID`)

all.dn <- unique(c(AM,CL,N2))

save(AM,CL,N2,all.dn,file = "data/dn.Rdata")
```

# Venn Diagram: differentially expressed genes

Visualize up-regulated genes in the three strains as a Venn diagram.
```{r venn-upregulated}
load("data/up.Rdata")

dir.create("results", showWarnings = FALSE)
filename = "results/2019-05-16-venn_tor_rnai_upregulated.tiff"
title = "Upregulated (p < 0.1)"

venn.diagram(
  x = list(AM, CL, N2),
  filename = filename,
  category = c("AM140","CL2006","N2"),
  resolution = 300,
  imagetype = "tiff",
  main = title,
  main.cex = 3,
  main.fontfamily = "sans",
  fill = viridis(3),
  cex = 2.5 ,# size of the areas’ labels
  fontfamily = "sans", # the fontfamily of the areas’ labels
  cat.cex = 2.5 ,#the size of the category names
  cat.pos = c(0,45,0),
  cat.dist = c(-.02,.05,-.04),
  cat.fontfamily = "sans"
)
```

Visualize down-regulated genes in the three strains.
```{r venn-downregulated}
load("data/dn.Rdata")

dir.create("results", showWarnings = FALSE)
filename = "results/2019-05-16-venn_tor_rnai_downregulated.tiff"
title = "Downregulated (p < 0.1)"

venn.diagram(
  x = list(AM, CL, N2),
  filename = filename,
  category = c("AM140","CL2006","N2"),
  resolution = 300,
  imagetype = "tiff",
  main = title,
  main.cex = 3,
  main.fontfamily = "sans",
  fill = viridis(3),
  cex = 2.5 ,# size of the areas’ labels
  fontfamily = "sans", # the fontfamily of the areas’ labels
  cat.cex = 2.5 ,#the size of the category names
  cat.pos = c(0,0,0),
  cat.dist = -0.05,
  cat.fontfamily = "sans"
)
```

Remove variables used for Venn diagram.
```{r clean-venn}
rm(list = c("filename","title"))
```

# Heatmap: differentially expressed genes

## Data-prep

Vector of both up- and down-regulated genes (*p*-value < 0.1) to plot in heatmap.
```{r pool-regulated}
load("data/up.Rdata")
load("data/dn.Rdata")
goi <- union(all.up,all.dn)
```

Create a fold-change values matrix of genes of interest. Import normalized counts of differentially expressed genes. Convert the counts to a numerical matrix, and names the genes.
```{r import-reads, results='hide'}
norm_df <- 
  read_csv("raw/2018-10-15-Results_files/normalized_counts.csv") %>%
  dplyr::rename(gene_id = X1) %>% 
  filter(gene_id %in% goi)

# Numerical Matrix of reads-only.
norm_mat <- as.matrix(norm_df[,5:27])

# Name the genes
rownames(norm_mat) <- norm_df$gene_id
```

Transform values to Z-score by gene (row), save the data matrix.
```{r, results='hide'}
z_mat <- 
  (norm_mat - rowMeans(norm_mat)) / apply(norm_mat, MARGIN = 1,FUN = sd) 
# Z-score by row

save(goi,norm_mat,z_mat,file = "data/heatmap.Rdata")
```

## Load Dataframes (**checkpoint**).
```{r load-data}
# load("data/heatmap.Rdata")
```

## Clustering

Hierarchical Clustering by row and column.
```{r h-clustering}
rc <- z_mat %>% dist %>% hclust %>% as.dendrogram
cc <- z_mat %>% t %>% dist %>% hclust %>% as.dendrogram
```

## Plot

Set colors for Heatmap, 256 colors in the scale.
```{r color-scale}
col <- rev(colorRampPalette(brewer.pal(10, "RdBu"))(16)) 
# Reverse so Blue is downregulated.

col.lab <- brewer.pal(6, "Paired")
```

Plot heatmap.
Based on [1](https://warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r/heatmap/) and [2](https://www.datanovia.com/en/lessons/heatmap-in-r-static-and-interactive-visualization/).
```{r plot-heatmap}
# dev.off()
svg(file = "results/2019-05-20-heatmap.svg",
                  width = 4, 
                  height = 3,
                  pointsize = 9)

# par$xpd=FALSE by default
par(xpd=TRUE)

heatmap.2(x = z_mat, # data

          # Colors
          colsep = c(8,15), # Separate columns by strain
          
          col = col, # cell colors
          # Color samples by treatment and strain
          ColSideColors = c(rep(col.lab[1], 4), 
                            rep(col.lab[2], 4), 
                            rep(col.lab[3], 3), 
                            rep(col.lab[4], 4), 
                            rep(col.lab[5], 4), 
                            rep(col.lab[6], 4)),
          trace = "none", # Remove the cyan trace on the cells
          density.info = "none", # Remove the density curve in legend
          
          # Sizes
          key = TRUE,
          keysize = 1, # Color legend key size
          key.par = list(cex = 0.5), # Color scale legend font
          lhei = c(1,4), # ColorScale-to-Heatmap ratio, height
          lwid=c(1,4), # ColorScale-to-Heatmap ratio, width
          
          # Labels
          labRow = "", # Label of rows
          labCol = "", # Label of rows
          srtCol = 30, # Angle column labels.
          key.title = NA, # Title of Color Scale
          key.xlab = "Gene Expression (Z-score)", # Color Scale X-label
          # main = "Genes Regulated by Torsin RNAi",
          ylab = "Genes",
          xlab = "Samples",
          
          # Clustering
          Rowv=rc, # row dendogram
          Colv=F, # don't reorder columns
          dendrogram = "row", # draw no dendogram
          
          margins = c(2,2)
          
          )

# Set legend
legend(x=.25, y=1.3, # Legend poisition
    legend = c( # Legend labels
      'N2 EV',
      'N2 tor-2',
      'AM140 EV',
      'AM140 tor-2',
      'CL2006 EV',
      'CL2006 tor-2'
    ),
    col = col.lab,# Label colors
    lty= 1, # Label line type
    lwd = 5, # Label line width
    cex=.7, # Label text size
    ncol = 3 # Organize into 3 columns
    )

dev.off()
```

# Bar graph: Oppositely regulated genes

## Load Dataframes (**checkpoint**).
```{r load-data}
# load("data/heatmap.Rdata")
# load("data/import-clean-ge.Rdata")
```

## Data-prep
```{r per-strain-foldchange}
fc <- 
  gepc_tested %>% 
  # Remove irrelevant columns and keep p-adjusted and FC per gene.
  dplyr::select(-baseMean,-pvalue,-flag) %>%
  gather(variable,value, -(strain:Gene_Name)) %>%
  unite(temp, strain, variable) %>%
  spread(temp, value) %>%
  
  # difference between CL2006 and AM140 FC, value and direction
  dplyr::mutate(diff = abs(AM140_log2FC-CL2006_log2FC),
                sign = sign(AM140_log2FC-CL2006_log2FC)) %>% 
  
  # remove 
  dplyr::filter(AM140_FDR_p < .1 | CL2006_FDR_p < .1 | N2_FDR_p < .05) %>%
  
  # order by difference between CL2006 and AM140 FC
  dplyr::arrange(desc(diff))

fc$sign[fc$sign == 1] <- sprintf('↑AM140 ↓CL2006')
fc$sign[fc$sign == -1] <- sprintf('↓AM140 ↑CL2006')
```

## Plot
```{r plot}
# dev.off()
svg(filename = "results/2019-05-21-response-AM140-CL2006.svg",
          width = 3, 
          height = 4.5)

am_up <- fc %>%
  dplyr::filter(sign =='↑AM140 ↓CL2006') %>%
  ggplot(
    aes(x = reorder(Gene_Name, -diff), # Order by difference size
        y = diff
    )
  ) + 
  geom_col() + 
  ylab(NULL) + # See above note for how calculated
  xlab(NULL) + 
  ggtitle('↑AM140 ↓CL2006') + 
  coord_flip() + # 90 degrees flip
  theme_classic() + 
  # Removes the space between the x-axis labels and zero values
  scale_y_continuous(expand = c(0,0)) +
  theme(text = element_text(size=7,color='black'),
        plot.margin = unit(c(0,1,0,0), 'lines')) # font of entire plot

am_dn <- fc %>%
  dplyr::filter(sign =='↓AM140 ↑CL2006') %>%
  ggplot(
    aes(x = reorder(Gene_Name, -diff), # Order by difference size
        y = diff
    )
  ) + 
  geom_col() + 
  ylab(NULL) + # See above note for how calculated
  xlab(NULL) + 
  ggtitle('↓AM140 ↑CL2006') +
  coord_flip() + # 90 degrees flip
  theme_classic() + 
  # Removes the space between the x-axis labels and zero values
  scale_y_continuous(expand = c(0,0)) +
  theme(text = element_text(size=7,color='black'),
        plot.margin = unit(c(0,1,0,0), 'lines')) # font of entire plot

figure <- ggarrange(am_up,am_dn,
                    ncol = 2, nrow = 1,
                    hjust=c(0,4))
annotate_figure(figure,
                bottom = text_grob("Response Difference",face="bold",size = 7))

dev.off()
```

# Gene Tables



# GO GSEA
Gene Ontology (GO) Gene Set Enrichment Analysis (GSEA)

## Load Dataframes (**checkpoint**).
```{r load-data}
# load("data/import-clean-ge.Rdata", verbose = TRUE)
# load("data/heatmap.Rdata", verbose = TRUE)
```

## Generate a Ranked Gene List
The ranked gene-list for the analysis needs to be ordered in a descending order by Fold-change. In the past, I tried ordering by *p*-values and by $-log10(p-value)*log_{2}(Fold-Change)$, but these didn't give meaningful results. I'll analyze the gene-lists from each strain independently. 
**Split this into 3 analyses**.

Create a list of vectors of the Gene IDs for all the identified genes, ranked.
**Note:** The analysis is done only to genes that passed the p-adjusted criteria. Need to do this to all the genes in the dataset.
```{r ranked}
ranked <- list()

strains = c("N2","AM140","CL2006")

for (strain in strains) {
  df <- gepc_tested %>% 
    dplyr::filter((!!strain) == strain) %>%
    dplyr::arrange(desc(log2FC))
  ranked[[eval(strain)]] <- setNames(df$log2FC,df$Gene_ID)
}
```

## Fetch attributes from BioMaRt
Load the biomart data
```{r load-biomart}
load("data/2019-01-24-GO-annotations.Rdata",verbose = TRUE)
```

No need to run if the biomart data loaded unchanged.
```{r biomart-object}
mart = useMart(biomart="ensembl", dataset="celegans_gene_ensembl")
```

Fetch Gene-Ontology (GO) and GO-Slim annotations for *C. elegans*.
```{r go-annotations}
# GO attributes of interest
att <- c("ensembl_gene_id",
         "go_id",
         "name_1006",
         "definition_1006",
         "go_linkage_type",
         "namespace_1003")
go_ann <- getBM(attributes = att, 
                filters=list(biotype='protein_coding'),
                mart=mart)

# Same for GO-slim attributes
att <- c("ensembl_gene_id",
         "goslim_goa_accession",
         "name_1006",
         "goslim_goa_description",
         "go_linkage_type",
         "namespace_1003")
gosl_ann <- getBM(attributes = att, 
                  filters=list(biotype='protein_coding'),
                  mart=mart)
```
**STopped here**
The process takes a while, so save the data for faster future run-times.
```{r save-annotations}
save(gosl_ann, file = "data/2019-05-21-GO-annotations.Rdata")
#save(go_ann,gosl_ann, file = "data/2019-01-24-GO-annotations.Rdata")
```

## GSEA
```{r load-annotations}
load("data/2019-05-21-GO-annotations.Rdata",verbose = TRUE)
```

View possible GO terms categories
```{r go-categories}
levels(factor(gosl_ann$namespace_1003))
```

```
## Parse GO terms for input to GSEA

The goal here is to separate the GO terms of the three GO categories ...
- biological_process
- molecular_function
- cellular_component

... and generate lists, in which each vector of gene IDs is named by the GO term and its accession number.
```

The "IEA"-labelled GO terms correspond to annotations that were automatically determined, with _no experimental support_. Therefore, it is excluded. **Consider keeping**
```{r iea-exclude}
gosl_ann <- filter(gosl_ann,go_linkage_type != "IEA")
```

Extract GO annotations that are under *Biological Process*.
```{r parse-bp}
bp <- go_ann %>% filter(namespace_1003 == "biological_process")

bp <- split(bp$ensembl_gene_id,paste(bp$go_id,bp$name_1006))
bp <- lapply(bp, unique) # Removing gene IDs that are repeated under a term.
```

Extract GO annotations that are under *Molecular Function*.
```{r parse-mf}
mf <- go_ann %>% filter(namespace_1003 == "molecular_function")

mf <- split(mf$ensembl_gene_id,paste(mf$go_id,mf$name_1006))
mf <- lapply(mf, unique) # Removing gene IDs that are repeated under a term.
```

Extract GO annotations that are under *Cellular Component*.
```{r parse-cc}
cc <- go_ann %>% filter(namespace_1003 == "cellular_component")

cc <- split(cc$ensembl_gene_id,paste(cc$go_id,cc$name_1006))
cc <- lapply(cc, unique) # Removing gene IDs that are repeated under a term.
```

## Perform GO-GSEA
Using the package **fgsea** to perform GSEA, focusing on medium-sized gene-sets (with 15-500 members), 10,000 permutations.
```{r gsea-parameters}
minSize <- 15
maxSize <- 500
nperm <- 10000
```

```{r results-summary}
res <- data.frame()
```

### BP
```{r BP-N2}
strain <- "N2"
pathways_name <- "Biological Process"
ranked <- ranked_n2
pathways <- bp

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

```{r BP-AM140}
strain <- "AM140"
pathways_name <- "Biological Process"
ranked <- ranked_am
pathways <- bp

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

```{r BP-CL2006}
strain <- "CL2006"
pathways_name <- "Biological Process"
ranked <- ranked_cl
pathways <- bp

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

### MF
```{r MF-N2}
strain <- "N2"
pathways_name <- "Molecular Function"
ranked <- ranked_n2
pathways <- mf

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

```{r MF-AM140}
strain <- "AM140"
pathways_name <- "Molecular Function"
ranked <- ranked_am
pathways <- mf

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

```{r MF-CL2006}
strain <- "CL2006"
pathways_name <- "Molecular Function"
ranked <- ranked_cl
pathways <- mf

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

### CC
```{r CC-N2}
strain <- "N2"
pathways_name <- "Cellular Component"
ranked <- ranked_n2
pathways <- cc

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

```{r CC-AM140}
strain <- "AM140"
pathways_name <- "Cellular Component"
ranked <- ranked_am
pathways <- cc

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

```{r CC-CL2006}
strain <- "CL2006"
pathways_name <- "Cellular Component"
ranked <- ranked_cl
pathways <- cc

set.seed(159159)
fgseaRes <- fgsea(pathways = pathways, 
                  stats = ranked,
                  minSize = minSize,
                  maxSize = maxSize,
                  nperm = nperm)

# Print top 5 enriched pathways
print(head(fgseaRes[order(pval),][padj < 0.05],5))

res <- fgseaRes[order(pval),][padj < 0.05] %>%
  mutate(strain = strain, GO_category = pathways_name) %>%
  bind_rows(res)
```

## Export GSEA Results

Save the results of the GSEA
```{r save-gsea}
save(res, 
     file = paste0("results/",today,"-gsea",".Rdata"))

# Convert the column of gene names that are associated to each GO term from 
# a list of strings, to a string.
res$leadingEdge <- sapply(res$leadingEdge,
                          FUN = function(x) paste(x, collapse = ' '))

# Reorder the columns
res <- res %>% dplyr::select(strain,GO_category,everything())
write_csv(x = res,path = paste0("results/",today,"-gsea",".csv"))
```

# Gene lists tables

Focus on differentially expressed genes and renname columns
```{r}
tab <- ge %>% 
  filter(`Gene ID` %in% goi) %>% 
  rename(Gene_Name = `gene name`,
         Gene_ID = `Gene ID`,
         log2FC = `log2FoldChange Treated vs untreated`,
         FDR_adj_pvalue = FDR_adj_pvalue) %>%
  select(-baseMean)
```

Combined table for all three strains, up- and down-regulated genes.
```{r}
tab %>% 
  gather(score,quant,log2FC,FDR_adj_pvalue) %>%
  unite(score,strain,score,sep = "_") %>%
  spread(score,quant) %>%
  select(Gene_ID,Gene_Name,N2_log2FC,N2_FDR_adj_pvalue,AM140_log2FC,AM140_FDR_adj_pvalue,
         CL2006_log2FC,CL2006_FDR_adj_pvalue) %>% # Reorder columns
  write_excel_csv(path = "results/gene_lists/genes_list.csv")
```

Table for each strain, up- or down-regulated, individualy.
```{r}
# N2 upregulated
tab %>% filter(strain == 'N2',
               FDR_adj_pvalue < 0.1,
               log2FC > 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/N2_upreg.csv")

# N2 downregulated
tab %>% filter(strain == 'N2',
               FDR_adj_pvalue < 0.1,
               log2FC < 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/N2_dnreg.csv")

# AM140 upregulated
tab %>% filter(strain == 'AM140',
               FDR_adj_pvalue < 0.1,
               log2FC > 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/AM140_upreg.csv")

# AM140 downregulated
tab %>% filter(strain == 'AM140',
               FDR_adj_pvalue < 0.1,
               log2FC < 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/AM140_dnreg.csv")
  
# CL2006 upregulated
tab %>% filter(strain == 'CL2006',
               FDR_adj_pvalue < 0.1,
               log2FC > 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/CL2006_upreg.csv")

# CL2006  downregulated
tab %>% filter(strain == 'CL2006',
               FDR_adj_pvalue < 0.1,
               log2FC < 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/CL2006_dnreg.csv")
```


```{r session-info}
sessionInfo()
```

