---
title: "Data import and cleaning from .xlsx files sent by the Technion"
author: "Amir Levine"
date: "2018-10-29"
output: md_document
---

## Set parameters
```{r parameters, results='hide'}
if (!require(BiocManager)) {install.packages("BiocManager"); library(BiocManager)}

today = Sys.Date()
results_dir = "../results/"
data_dir = "../data/"
raw_data_dir = "../raw"
```


## Data Cleaning

Load tidyverse.
```{r packages, results='hide'}
# Standalone R
# if (!require(tidyverse)) {install.packages("tidyverse"); library(tidyverse)}

# Conda R
library(tidyverse)
# Package needs to be install through the conda package manager
```

Import the gene expression data sent by Efrat from the Technion.
Specifiy how to treat NA values.
Specify the columns' data-types, as by default in loads all as characters.
```{r import-data}
AM <- readxl::read_excel(
  path = "../raw/2018-10-25-Results_files/Final_results_AM140_treated_vs_untreated_.xlsx", 
  col_names = TRUE, 
  col_types = c(rep("text",4),rep("numeric",5),rep("text",4)), 
  na = 'NA')
CL <- readxl::read_excel(
  path = "../raw/2018-10-25-Results_files/Final_results_CL2006_treated_vs_untreated_.xlsx",
  col_names = TRUE, 
  col_types = c(rep("text",4),rep("numeric",5),rep("text",4)), 
  na = 'NA')
N2 <- readxl::read_excel(
  path = "../raw/2018-10-25-Results_files/Final_results_N2_treated_vs_untreated.xlsx",
  col_names = TRUE, 
  col_types = c(rep("text",4),rep("numeric",5),rep("text",4)), 
  na = 'NA')
```

Merge the three strain specific dataframes to one gene expression (ge) dataframe.
Remove unnecessary columns for further analysis.
Focus on genes that underwent the multiple comparisons test (*Tested*).
Sort the dataframe by the adjusted p-value.
```{r process-data}
ge <- bind_rows(list(N2 = N2,AM140 = AM, CL2006 = CL), .id = 'strain') %>%
  dplyr::filter(`flag Treated vs untreated` == "Tested") %>%
  dplyr::select(strain,
                `Gene ID`,
                `gene name`,
                baseMean,
                `log2FoldChange Treated vs untreated`,
                `padj Treated vs untreated`) %>%
  arrange(`padj Treated vs untreated`) %>%
  group_by(strain)
```

Save cleaned dataframes and remove individual strains data-frames (save checkpoint).
```{r save-data}
save(AM,CL,N2,ge,
     file = "../data/import-clean-ge.Rdata")
rm(list = c("AM","CL","N2"))
```

## Data preparation

Load the clean dataframes (load checkpoint).
```{r load-data}
# load("data/import-clean-ge.Rdata")
```

Obtain gene lists of upregulated genes (*p*-value < 0.1), for each strain.
Create a gene-name vector to generate a heatmap.
Save data.
```{r get-upregulated}
AM <- ge %>% 
  filter(strain == "AM140",
         `padj Treated vs untreated` < 0.1,
         `log2FoldChange Treated vs untreated` > 0) %>%
  pull(`Gene ID`)

CL <- ge %>% 
  filter(strain == "CL2006",
         `padj Treated vs untreated` < 0.1,
         `log2FoldChange Treated vs untreated` > 0) %>%
  pull(`Gene ID`)

N2 <- ge %>% 
  filter(strain == "N2",
         `padj Treated vs untreated` < 0.1,
         `log2FoldChange Treated vs untreated` > 0) %>%
  pull(`Gene ID`)

all.up <- unique(c(AM,CL,N2))

save(AM,CL,N2,all.up,file = "../data/up.Rdata")
```

Obtain gene lists of downregulated genes (*p*-value < 0.1), for each strain.
Combine all genes for heatmap.
```{r get-downregulated}
AM <- ge %>% 
  filter(strain == "AM140",
         `padj Treated vs untreated` < 0.1,
         `log2FoldChange Treated vs untreated` < 0) %>%
  pull(`Gene ID`)

CL <- ge %>% 
  filter(strain == "CL2006",
         `padj Treated vs untreated` < 0.1,
         `log2FoldChange Treated vs untreated` < 0) %>%
  pull(`Gene ID`)

N2<- ge %>% 
  filter(strain == "N2",
         `padj Treated vs untreated` < 0.1,
         `log2FoldChange Treated vs untreated` < 0) %>%
  pull(`Gene ID`)

all.dn <- unique(c(AM,CL,N2))

save(AM,CL,N2,all.dn,file = "../data/dn.Rdata")
```

## Venn Diagram

Load packages
```{r packages, results='hide'}
# Standalone
if (!require(VennDiagram)) {install.packages("VennDiagram"); library(VennDiagram)}
if (!require(viridis)) {install.packages("viridis"); library(viridis)}

# Conda
library(VennDiagram)
library(viridis)
```

Visualize up-regulated genes in the three strains as a Venn diagram.
```{r venn-upregulated}
load("../data/up.Rdata")

dir.create("../results", showWarnings = FALSE)
filename = "../results/2018-11-04-venn_anc1_upregulated.tiff"
title = "Upregulated (p < 0.1)"

venn.diagram(
  x = list(AM, CL, N2),
  filename = filename,
  category = c("AM140","CL2006","N2"),
  resolution = 300,
  imagetype = "tiff",
  main = title,
  main.cex = 3,
  main.fontfamily = "sans",
  fill = viridis(3),
  cex = 2.5 ,# size of the areas’ labels
  fontfamily = "sans", # the fontfamily of the areas’ labels
  cat.cex = 2.5 ,#the size of the category names
  cat.pos = c(0,45,0),
  cat.dist = c(-.02,.05,-.04),
  cat.fontfamily = "sans"
)
```

Visualize down-regulated genes in the three strains.
```{r venn-downregulated}
load("../data/dn.Rdata")

dir.create("../results", showWarnings = FALSE)
filename = "../results/2018-11-04-venn_anc1_downregulated.tiff"
title = "Downregulated (p < 0.1)"

venn.diagram(
  x = list(AM, CL, N2),
  filename = filename,
  category = c("AM140","CL2006","N2"),
  resolution = 300,
  imagetype = "tiff",
  main = title,
  main.cex = 3,
  main.fontfamily = "sans",
  fill = viridis(3),
  cex = 2.5 ,# size of the areas’ labels
  fontfamily = "sans", # the fontfamily of the areas’ labels
  cat.cex = 2.5 ,#the size of the category names
  cat.pos = c(0,0,0),
  cat.dist = -0.05,
  cat.fontfamily = "sans"
)
```

Remove variables used for Venn diagram.
```{r clean-venn}
rm(list = c("filename","title"))
```

## Heatmap of differentially expressed genes

Load pacakges
```{r packages, results='hide'}
# Standalone
if (!require(gplots)) {install.packages("gplots"); library(gplots)}
if (!require(RColorBrewer)) {install.packages("RColorBrewer"); library(RColorBrewer)}
# Conda
library(RColorBrewer)
```

Vector of both up- and down-regulated genes (*p*-value < 0.1) to plot in heatmap.
```{r pool-regulated}
load("../data/up.Rdata")
load("../data/dn.Rdata")
goi <- union(all.up,all.dn)
```

Create a fold-change values matrix of genes of interest.
Import normalized counts of differentially expressed genes.
Covert the counts to a numerical matrix, and names the genes.
```{r import-reads, results='hide'}
norm <- read_csv("../raw/2018-10-15-Results_files/normalized_counts.csv") %>% 
  rename("gene_id" = X1) %>% filter(gene_id %in% goi)

# Numerical Matrix of reads-only.
norm_mat <- as.matrix(norm[,5:27])

# Name the genes
rownames(norm_mat) <- norm$gene_id
```

Transform values to Z-score by gene (row)
```{r, results='hide'}
z_mat <- (norm_mat - rowMeans(norm_mat)) / apply(norm_mat, MARGIN = 1,FUN = sd) # Z-score by row

save(goi,norm_mat,z_mat,file = "../data/heatmap.Rdata")
```

Clustering
```{r h-clustering}
rc <- z_mat %>% dist %>% hclust %>% as.dendrogram
cc <- z_mat %>% t %>% dist %>% hclust %>% as.dendrogram
```

Set colors for Heatmap, 256 colors in the scale.
```{r color-scale}
col <- rev(colorRampPalette(brewer.pal(10, "RdBu"))(256)) # Reverse so Blue is downregulated.
col.lab <- brewer.pal(6, "Paired")
```

Plot heatmap
https://warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r/heatmap/
https://www.datanovia.com/en/lessons/heatmap-in-r-static-and-interactive-visualization/
```{r plot-heatmap}
# dev.off()
svg(file = paste0(results_dir,today,"-heatmap.svg",
                  width = 8.5, 
                  height = 11.5,
                  pointsize = 7))

heatmap.2(x = z_mat, # data

          # Colors
          colsep = c(8,15), # Separate columns by strain
          
          col = col, # cell colors
          # Color samples by treatment and strain
          ColSideColors = c(rep(col.lab[1], 4), 
                            rep(col.lab[2], 4), 
                            rep(col.lab[3], 3), 
                            rep(col.lab[4], 4), 
                            rep(col.lab[5], 4), 
                            rep(col.lab[6], 4)),
          trace = "none", # Remove the cyan trace on the cells
          density.info = "none", # Remove the density curve in legend
          
          # Sizes
          keysize = 1.05, # Color legend key size
          key.par = list(cex = 0.7), # Color scale legend font
          
          # Labels
          labRow = "", # Label of rows
          srtCol = 30, # Angle column labels.
          key.title = "Gene Expression (Z-score)",
          main = "Genes Regulated by Torsin RNAi",
          ylab = "Genes",
          xlab = "Samples",
          
          # Clustering
          Rowv=rc, # row dendogram
          Colv=F, # don't reorder columns
          dendrogram = "row", # draw only row dendogram
          
          margins = c(6,2)
          
          )

dev.off()
```

# Analyze gene annotations

## Fetch attributes from BioMaRt
```{r packages}
if (!require(biomaRt)) {BiocManager::install("biomaRt"); library(biomaRt)}
if (!require(org.Ce.eg.db)) {BiocManager::install("org.Ce.eg.db"); library(org.Ce.eg.db)}
```


## Gene Set Enrichment Analysis (GSEA)
Load packages
```{r packages, results='hide'}

```



## Gene lists tables

Focus on differentially expressed genes and renname columns
```{r}
tab <- ge %>% 
  filter(`Gene ID` %in% goi) %>% 
  rename(Gene_Name = `gene name`,
         Gene_ID = `Gene ID`,
         log2FC = `log2FoldChange Treated vs untreated`,
         FDR_adj_pvalue = `padj Treated vs untreated`) %>%
  select(-baseMean)
```


Combined table for all three strains, up- and down-regulated genes.
```{r}
tab %>% 
  gather(score,quant,log2FC,FDR_adj_pvalue) %>%
  unite(score,strain,score,sep = "_") %>%
  spread(score,quant) %>%
  select(Gene_ID,Gene_Name,N2_log2FC,N2_FDR_adj_pvalue,AM140_log2FC,AM140_FDR_adj_pvalue,
         CL2006_log2FC,CL2006_FDR_adj_pvalue) %>% # Reorder columns
  write_excel_csv(path = "results/gene_lists/genes_list.csv")
```

Table for each strain, up- or down-regulated, individualy.
```{r}
# N2 upregulated
tab %>% filter(strain == 'N2',
               FDR_adj_pvalue < 0.1,
               log2FC > 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/N2_upreg.csv")

# N2 downregulated
tab %>% filter(strain == 'N2',
               FDR_adj_pvalue < 0.1,
               log2FC < 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/N2_dnreg.csv")

# AM140 upregulated
tab %>% filter(strain == 'AM140',
               FDR_adj_pvalue < 0.1,
               log2FC > 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/AM140_upreg.csv")

# AM140 downregulated
tab %>% filter(strain == 'AM140',
               FDR_adj_pvalue < 0.1,
               log2FC < 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/AM140_dnreg.csv")
  
# CL2006 upregulated
tab %>% filter(strain == 'CL2006',
               FDR_adj_pvalue < 0.1,
               log2FC > 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/CL2006_upreg.csv")

# CL2006  downregulated
tab %>% filter(strain == 'CL2006',
               FDR_adj_pvalue < 0.1,
               log2FC < 0) %>%
  arrange(FDR_adj_pvalue) %>%
  write_excel_csv(path = "results/gene_lists/CL2006_dnreg.csv")
```



## GO/GSEA analysis
https://bioconductor.org/packages/release/bioc/html/topGO.html
https://bioconductor.org/packages/3.7/bioc/html/clusterProfiler.html
http://bioconductor.org/packages/release/bioc/html/gage.html
http://software.broadinstitute.org/gsea/msigdb/collections.jsp
https://bioconductor.org/packages/release/bioc/html/GSEABase.html



```{r}
write(ge %>% filter(`padj Treated vs untreated` < 0.1) %>% pull(`Gene ID`) %>% unique(),
      file = "a.txt")
```



